app/controllers/search_controller.rb

```diff
@@ -41,47 +39,60 @@ def new
   private
 
   def load_vars
-    @search_query = params[:q] || ''
+    query = params[:q].strip || ''
```

Comment:
you can use 'fetch' to set the default value here if you want:

query = params.fetch(:q) { '' }